src/query/src/dist_plan/analyzer.rs

```diff
@@ -381,38 +417,85 @@ impl PlanRewriter {
 /// Requirements enforced by this rewriter:
 /// - Enforce column requirements for `LogicalPlan::Projection` nodes. Makes sure the
 ///   required columns are available in the sub plan.
+///
+#[derive(Debug)]
 struct EnforceDistRequirementRewriter {
-    column_requirements: HashSet<Column>,
+    // only enforce column requirements after the expanding node in question
```

Comment:
What does "the expanding node in question" mean?

Reply:
> What does "the expanding node in question" mean?

the node in `self.expand(node)`